plugins/prometheus/debug.go

```diff
@@ -0,0 +1,156 @@
+package prometheus
+
+import (
+	"sync/atomic"
+	"unsafe"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/gohornet/hornet/pkg/snapshot"
+	"github.com/gohornet/hornet/pkg/whiteflag"
+	"github.com/iotaledger/hive.go/events"
+)
+
+var (
+	snapshotBuckets     = []float64{.1, .2, .5, 1, 2, 5, 10, 20, 50, 100, 200, 500}
+	pruningBuckets      = []float64{.1, .2, .5, 1, 2, 5, 10, 20, 50, 100, 200, 500}
+	confirmationBuckets = []float64{.005, .01, .025, .05, .1, .25, .5, 1, 2.5, 5, 10, 20, 50, 100}
+
+	snapshotTotalDuration              prometheus.Histogram
+	snapshotDurations                  *prometheus.GaugeVec
+	databasePruningTotalDuration       prometheus.Histogram
+	databasePruningDurations           *prometheus.GaugeVec
+	milestoneConfirmationTotalDuration prometheus.Histogram
+	milestoneConfirmationDurations     *prometheus.GaugeVec
+
+	lastSnapshotMetricsPointer        unsafe.Pointer
+	lastDatabasePruningMetricsPointer unsafe.Pointer
+	lastConfirmationMetricsPointer    unsafe.Pointer
```

Comment:
Remove the `unsafe.Pointer`s and use locks or anything else to deal with the data race.

Reply:
Fixed.